### PR TITLE
Dtspb 3371

### PR DESCRIPTION
**Pull request by DelythJustice**
### JIRA link (if applicable) ###
DTSPB-3371 resend iron mountain

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -2,5 +2,8 @@
   {"CaseTypeID": "LegacySearch", "CaseEventID": "createDummy", "UserRole": "caseworker-probate-issuer", "CRUD": "CRUD"},
   {"CaseTypeID": "LegacySearch", "CaseEventID": "legacySearch", "UserRole": "caseworker-probate-issuer", "CRUD": "CRUD"},
   {"CaseTypeID": "LegacySearch", "CaseEventID": "createDummy", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"},
-  {"CaseTypeID": "LegacySearch", "CaseEventID": "legacySearch", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"}
+  {"CaseTypeID": "LegacySearch", "CaseEventID": "legacySearch", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"},
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseEvent.json:5]: I suggest we take this opportunity to remove CU access for ordinary users for create dummy event
And make it only CRUD for superusers

File: ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseField.json [hunk 2]
```diff
@@ -16,5 +16,15 @@
   {"CaseTypeID": "LegacySearch", "CaseFieldID": "deceasedDateOfBirth", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"},
   {"CaseTypeID": "LegacySearch", "CaseFieldID": "deceasedDateOfDeath", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"},
   {"CaseTypeID": "LegacySearch", "CaseFieldID": "legacySearchResultRows", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"},
-  {"CaseTypeID": "LegacySearch", "CaseFieldID": "doNotSaveCase", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"}
+  {"CaseTypeID": "LegacySearch", "CaseFieldID": "doNotSaveCase", "UserRole": "caseworker-probate-caseadmin", "CRUD": "CRUD"},
+  {"CaseTypeID": "LegacySearch", "CaseFieldID": "customHistoryViewer", "UserRole": "caseworker-probate-superuser", "CRUD": "R"},
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseField.json:20]: some of these are already setup on the authcasefield for superuser, eg. deceasedSurname

File: ccdImports/configFiles/CCD_Probate_Legacy_Search/CaseEvent.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
...
-  {"LiveFrom": "01/01/2017", "CaseTypeID": "LegacySearch", "ID": "legacySearch", "Name": "Search for legacy cases", "Description": "Search for legacy cases", "DisplayOrder": 2, "PreConditionState(s)": "*", "PostConditionState": "*", "CallBackURLAboutToSubmitEvent": "http://${CCD_DEF_CASE_SERVICE_BASE_URL}/legacy/resetSearch", "SecurityClassification": "Public", "ShowSummary": "Y"}
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "LegacySearch", "ID": "legacySearch", "Name": "Search for legacy cases", "Description": "Search for legacy cases", "DisplayOrder": 2, "PreConditionState(s)": "*", "PostConditionState": "*", "CallBackURLAboutToSubmitEvent": "http://${CCD_DEF_CASE_SERVICE_BASE_URL}/legacy/resetSearch", "SecurityClassification": "Public", "ShowSummary": "Y"},
+  {"LiveFrom": "16/03/2023", "CaseTypeID": "LegacySearch", "ID": "resendData", "Name": "Resend Data", "Description": "Resend Iron Mountain Data feed", "DisplayOrder": 3, "PreConditionState(s)": "*", "PostConditionState": "*", "CallBackURLAboutToSubmitEvent": "http://${CCD_DEF_CASE_SERVICE_BASE_URL}/data-extract/resend-iron-mountain", "SecurityClassification": "Public", "ShowSummary": "Y"}
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Legacy_Search/CaseEvent.json]: can we rename to dataExtractResendDate - I assume we will use the same field for any type of future data extract resend

File: src/cftlib/java/uk/gov/hmcts/probate/CftLibConfig.java [hunk 4]
```diff
@@ -22,6 +22,12 @@ public void configure(CFTLib lib) throws Exception {
         lib.createProfile("probatesolicitortestorgtest1@gmail.com", "PROBATE", "Caveat", "Pending");
         lib.createProfile("probatecaseworker@gmail.com", "PROBATE", "GrantOfRepresentation", "Pending");
         lib.createProfile("probatecaseworker@gmail.com", "PROBATE", "Caveat", "Pending");
+        lib.createProfile("probatesolicitortestorgtest1@gmail.com", "PROBATE", "LegacySearch", "Pending");
+        lib.createProfile("probatesolicitortestorgtest1@gmail.com", "PROBATE", "Caveat", "Pending");
+        lib.createProfile("probatecaseworker@gmail.com", "PROBATE", "GrantOfRepresentation", "Pending");
+        lib.createProfile("probatecaseworker@gmail.com", "PROBATE", "Caveat", "Pending");
+        lib.createProfile("probatebackoffice@gmail.com", "PROBATE", "LegacySearch", "Pending");
```
Comment (sanjayparekh-hmcts) [inline src/cftlib/java/uk/gov/hmcts/probate/CftLibConfig.java]: I dont think there is a Pending (default) state for LegacySearch case types

File: src/main/java/uk/gov/hmcts/probate/controller/DataExtractController.java [hunk 5]
```diff
@@ -67,6 +75,23 @@ public ResponseEntity initiateIronMountainExtract(@Parameter(name = "Date to fin
...
+    @Operation(summary = "Initiate IronMountain data extract with date",
+        description = "Date MUST be in callbackRequest 'yyyy-MM-dd'")
+    @PostMapping(path = "/resend-iron-mountain", consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity<AfterSubmitCallbackResponse> initiateIronMountainExtract(
+        @RequestBody CallbackRequest callbackRequest) throws NotificationClientException {
+
+        String resendDate = callbackRequest.getCaseDetails().getData().getResendDate();
+        dataExtractDateValidator.dateValidator(resendDate);
+
+        log.info("Calling resend perform Iron Mountain data extract from date...");
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/controller/DataExtractController.java]: put date in the logs too?

**Review by sanjayparekh-hmcts — DISMISSED** [review]

File: src/cftlib/java/uk/gov/hmcts/probate/CftLibConfig.java (hunk 4 above)
Comment (DelythJustice) [reply src/cftlib/java/uk/gov/hmcts/probate/CftLibConfig.java]: I'll just take these out as for local non aat

File: ccdImports/configFiles/CCD_Probate_Legacy_Search/CaseEvent.json (hunk 3 above)
Comment (DelythJustice) [reply ccdImports/configFiles/CCD_Probate_Legacy_Search/CaseEvent.json]: changing from resendData to dataExtractResendData (assuming that was a typo)

File: ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseField.json (hunk 2 above)
Comment (DelythJustice) [reply ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseField.json:20]: discussed, they are not

File: ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseEvent.json (hunk 1 above)
Comment (DelythJustice) [reply ccdImports/configFiles/CCD_Probate_Legacy_Search/AuthorisationCaseEvent.json:5]: talked to stephen about this

File: charts/probate-back-office/values.yaml [hunk 6]
```diff
@@ -69,7 +69,7 @@ java:
           alias: smeeAndFordEmail
         - name: ftpSignature
           alias: ftp.client.signature
-        - name: ftpEnv
+        - name: ftftpEnvpEnv
```
Comment (FeliTam) [inline charts/probate-back-office/values.yaml]: What is this change about?

Comment (DelythJustice) [reply charts/probate-back-office/values.yaml]: removed accidental change, thanks

**Review by FeliTam — APPROVED** [review]

**Pull request merged**